test/e2e/test_authcheck.go

```diff
@@ -0,0 +1,126 @@
+/*
+Copyright 2019 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"context"
+	"strings"
+	"testing"
+
+	reconcilertesting "github.com/google/knative-gcp/pkg/reconciler/testing"
+	"github.com/google/knative-gcp/test/lib"
+)
+
+const (
+	authCheckServiceAccountName = "auth-check-test"
+	fakeServiceAccount          = "fakeserviceaccount@test-project.iam.gserviceaccount.com"
+	fakeValue                   = "{\n  \"type\": \"service_account\",\n  \"project_id\": \"fake-project-id\",\n  \"private_key_id\": \"fake-key-id\"}"
+)
+
+func AuthCheckForPodCheckTestImpl(t *testing.T, authConfig lib.AuthConfig) {
+	ctx := context.Background()
+	t.Helper()
```

Comment:
I would prefer this is not here. As far as I can tell, all it does is cause the returned error line to not show anything in this function or below, instead return the line where this function is called.

As this function really is the test, I think it makes debugging harder when we don't quickly see the line of the failure.